src/univariate/discrete/poisson.jl

```diff
@@ -84,6 +84,11 @@ function entropy(d::Poisson{T}) where T<:Real
     end
 end
 
+function kldivergence(p::Poisson, q::Poisson)
+    λp = rate(p)
+    λq = rate(q)
+    return λq - λp + xlogx(λp) - xlogy(λp, λq)
```

Comment:
What's the reason for not using
```suggestion
    return λq - λp + xlogx(λp, λp / λq)
```
? This would probably be faster.

Reply:
That handle the case where lambdaq is 0

Reply:
:+1: Ah yes I missed that `xlogy(0, 0 / 0) === NaN`. The following would handle the case `λp = λq = 0` as well with only one `log` (`false` is a strong zero, i.e., `false * NaN = 0.0`):
```suggestion
    # `false` is a strong zero and ensures that `λp = 0` is handled correctly
    # we don't use `xlogy` since it returns `NaN` for `λp = λq = 0`
    return λq - λp + (λp > 0) * (λp * log(λp / λq))
```
In any case, we should add tests for these special cases to avoid that regressions are introduced accidentally.